code/__HELPERS/icon_smoothing.dm

```diff
@@ -41,18 +41,61 @@
 		if(!AM.anchored)
 			return 0
 
-		for(var/direction in alldirs)
+		for(var/direction in cardinal)
 			AM = find_type_in_direction(A, direction)
-			if(istype(AM))
-				if(AM.anchored)
-					adjacencies |= 1 << direction
-			else
-				if(AM)
-					adjacencies |= 1 << direction
+			if(istype(AM) && AM.anchored)
+				adjacencies |= 1 << direction
+			else if(AM)
+				adjacencies |= 1 << direction
```

Comment:
Well there's a problem with the code then, it does `if(!AM.anchored)` when AM may not be a movable and anchored is defined for movable only, which mean the proc will runtime if it's not a movable.